sw/host/opentitanlib/src/util/parse_int.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::num::ParseIntError;
+
+/// Trait for parsing integers.
+///
+/// Strings beginning with the common and classic prefixes `0x`, `0o`, `0b` or `0` are parsed
+/// as hex, octal, binary, and octal (classic).  Anything else is parsed as decimal.
+/// A leading `+` or `-` is permitted.  Any string parsed by `strtol(3)` or `strtoul(3)`
+/// will be parsed successfully.
+///
+/// This code was shamelessly adapted from https://github.com/Rust-for-Linux/linux/pull/87
+pub trait ParseInt: Sized {
+    fn from_str_radix(src: &str, radix: u32) -> Result<Self, ParseIntError>;
+
+    fn from_str(src: &str) -> Result<Self, ParseIntError> {
+        let (val, negative) = if let Some(v) = src.strip_prefix("-") {
+            (v, true)
+        } else if let Some(v) = src.strip_prefix("+") {
+            (v, false)
+        } else {
+            (src, false)
+        };
+
+        let (radix, digits) = if let Some(n) = val.strip_prefix("0x") {
+            (16, n)
+        } else if let Some(n) = val.strip_prefix("0x") {
+            (16, n)
+        } else if let Some(n) = val.strip_prefix("0o") {
+            (8, n)
+        } else if let Some(n) = val.strip_prefix("0O") {
+            (8, n)
+        } else if let Some(n) = val.strip_prefix("0b") {
+            (2, n)
+        } else if let Some(n) = val.strip_prefix("0B") {
+            (2, n)
+        } else if val.starts_with("0") {
+            (8, val)
+        } else {
+            (10, val)
+        };
```

Comment:
Thoughts?

```rust
match val.get(0..2) {
  "0x" | "0X" => (16, &n[2..]), // And so on.
  _ if val.starts_with("0") => (8, val),
  _ => (10, val),
}
```
Most of the reason to do this is to fold the capitalizations together.

Reply:
That's so much better.  Done.